modules/compiler/utils/include/compiler/utils/builtin_info.h

```diff
@@ -518,21 +587,55 @@ class BuiltinInfo {
            ID == eMuxBuiltinDMAWrite3D;
   }
 
+  /// @brief Gets information about a mux group operation builtin
+  ///
+  /// Note: Does not set the 'function' or 'type' members of the
+  /// GroupCollective.
+  ///
+  /// FIXME: This matches an equivalent function in group_collective_helpers.h
+  /// which runs on OpenCL builtins. Remove that once the transition to mux
+  /// builtins is complete.
+  static std::optional<GroupCollective> isMuxGroupCollective(BuiltinID ID);
+
   /// @brief Maps a ComputeMux builtin ID to its function name.
-  static llvm::StringRef getMuxBuiltinName(BuiltinID ID);
+  static std::string getMuxBuiltinName(
+      BuiltinID ID, llvm::ArrayRef<llvm::Type *> OverloadInfo = {});
+
+  /// @brief Mangles a type using the LLVM intrinsic scheme
+  ///
+  /// This is an extremely simple mangling scheme matching LLVM's intrinsic
+  /// mangling system. It is only designed to be used with a specific set of
+  /// types and is not a general-purpose mangler.
+  ///
+  /// * iXXX -> iXXX
+  /// * half -> f16
+  /// * float -> f32
+  /// * double -> f64
+  /// * <N x Ty> -> vNTy
+  /// * <vscale x N x Ty> -> nxvNTy
+  static std::string getMangledTypeStr(llvm::Type *Ty);
+
+  /// @brief Demangles a type using the LLVM intrinsic scheme - returns nullptr
+  /// if it was unable to demangle a type.
+  ///
+  /// @see getMangledTypeStr
+  static std::pair<llvm::Type *, llvm::StringRef> getDemangledTypeFromStr(
+      llvm::StringRef TyStr, llvm::LLVMContext &Ctx);
 
   /// @brief Defines the body of a ComputeMux builtin declaration
   ///
   /// If the Module already has a function definition with the corresponding
   /// function name, it is left alone and returned.
   ///
   /// Will declare any builtins it requires as transitive dependencies.
-  llvm::Function *defineMuxBuiltin(BuiltinID, llvm::Module &M);
+  llvm::Function *defineMuxBuiltin(
+      BuiltinID, llvm::Module &M,
+      llvm::ArrayRef<llvm::Type *> OverloadInfo = {});
```

Comment:
I think it would be useful to have an @param for the overloadInfo to give an idea of what it's for

Reply:
In fact, I don't see it really explained anywhere include the documentation (which looking at doesn't go into too much direct detail about builtinInfo - although talks about it quite a bit indirectly).

Reply:
I'd added a `@param` now, and have also added a static function `BuiltinInfo::isOverloadableMuxBuiltinID` to tie those concepts together

You're right that `BuiltinInfo` isn't very well documented in the docs. That's something we should address